core-4/src/main/java/scanning/parkingLot/FreeParkingLot.java

```diff
@@ -0,0 +1,27 @@
+package scanning.parkingLot;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.stereotype.Component;
+import scanning.car.Car;
+
+import java.util.List;
+
+@Component(value = "freeParkingLot")
+public class FreeParkingLot implements ParkingLot{
+
+    private String[] allCars;
+
+    @Autowired
+    public FreeParkingLot(ApplicationContext applicationContext){
```

Comment:
not bad, but we don't need any application context to be involved manually, try to find another solution

and Cars should be there too, this is a parking lot